src/js/stores/document.js

```diff
@@ -306,16 +306,20 @@ define(function (require, exports, module) {
          * Reset the given layer models.
          *
          * @private
-         * @param {{documentID: number, layers: Immutable.Iterable.<{layerID: number, descriptor: object}>}} payload
+         * @param {object} payload
+         * @param {number} payload.documentID
+         * @param {Immutable.Iterable.<{layerID: number, descriptor: object}>} payload.layers
+         * @param {boolean=} payload.suppressDirty
          */
         _handleLayerReset: function (payload) {
             var documentID = payload.documentID,
                 layerObjs = payload.layers,
                 document = this._openDocuments[documentID],
                 nextLayers = document.layers.resetLayers(layerObjs, document),
-                nextDocument = document.set("layers", nextLayers);
+                nextDocument = document.set("layers", nextLayers),
+                suppressDirty = payload.suppressDirty;
 
-            this.setDocument(nextDocument, true);
+            this.setDocument(nextDocument, !suppressDirty);
```

Comment:
thanks for fixing this.  I'm writing it down to document that mad review skillz.